src/category_theory/sites/closed.lean

```diff
@@ -0,0 +1,318 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.sites.sheaf_of_types
+import order.closure
+
+/-!
+# Closed sieves
+
+A natural closure operator on sieves is a closure operator on `sieve X` for each `X` which commutes
+with pullback.
+We show that a Grothendieck topology `J` induces a natural closure operator, and define what the
+closed sieves are. The collection of `J`-closed sieves forms a presheaf which is a sheaf for `J`,
+and further this presheaf can be used to determine the Grothendieck topology from the sheaf
+predicate.
+Finally we show that a natural closure operator on sieves induces a Grothendieck topology, and hence
+that natural closure operators are in bijection with Grothendieck topologies.
+
```

Comment:
Would you mind adding
```
## Main declarations
```
plus a list of what you think are the names of the most important defs/results in this file? If you use the fully qualified name (so `category_theory.grothendieck_topology.foobar`) then this will turn into a link in the documentation, so people can quickly navigate from the top of the file to the interesting declarations.

Reply:
I added some stuff - is this alright?

Reply:
Typically, in other files it is a list of the form
```
* `some.declaration.name`: short description
* `another.declaration`: another description
```
Sorry, I should have been a bit more clear about the intended format upfront.

Reply:
Contentwise it looks fine!